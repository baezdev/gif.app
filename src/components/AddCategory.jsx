import { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [value, setValue] = useState("");

  const handleInputChange = ({ target }) => setValue(target.value);

  const handleAddCategory = (e) => {
    e.preventDefault();

    const valueCategory = value.trim();
    if (valueCategory.length <= 3) {
      return;
    }
    onNewCategory(valueCategory);
    setValue("");
  };

  return (
    <form className="max-w-xl mx-auto my-12" onSubmit={handleAddCategory}>
      <div className="p-1 bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500">
        <input
          type="text"
          placeholder="Buscar gifs"
          value={value}
          onChange={handleInputChange}
          className="w-full text-3xl font-bold focus:outline-none py-3 h-full px-4 text-center bg-white"
        />
      </div>
    </form>
  );
};
