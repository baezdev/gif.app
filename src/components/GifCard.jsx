import PropTypes from "prop-types";

export const GifCard = ({ title, url }) => {
  return (
    <div className="w-full h-fit p-px bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500">
      <div>
        <img src={url} alt={title} className="w-full" />
        <p className="bg-slate-100 px-4 py-7 text-base text-center font-bold">
          {title}
        </p>
      </div>
    </div>
  );
};

GifCard.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
