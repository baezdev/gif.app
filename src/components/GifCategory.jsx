import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifCard } from "./";

export const GifCategory = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className="mb-10">
      <h3 className="text-4xl font-extrabold mb-5">{category}</h3>
      {isLoading && <p className="text-2xl font-bold">Cargando...</p>}
      <section className="grid grid-cols-3 gap-5">
        {images.map((gif) => (
          <GifCard {...gif} key={gif.id} />
        ))}
      </section>
    </div>
  );
};
