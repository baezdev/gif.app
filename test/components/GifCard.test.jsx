import { render, screen } from "@testing-library/react";
import { GifCard } from "../../src/components/GifCard";

describe("Pruebas en <GifCard />", () => {
  const title = "Goku";
  const url = "https:";

  test("match con el snapshot", () => {
    const { container } = render(<GifCard title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el url y el alt indicado", () => {
    render(<GifCard title={title} url={url} />);
    /* screen.debug(); */
    /* expect(screen.getByRole("img").src).toBe(url);
    expect(screen.getByRole("img").alt).toBe(title); */
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe de mostrar el titulo en el componente", () => {
    render(<GifCard title={title} url={url} />);

    expect(screen.getByText(title)).toBeTruthy();
  });
});
