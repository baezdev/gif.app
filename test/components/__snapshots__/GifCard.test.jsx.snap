// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard /> match con el snapshot 1`] = `
<div>
  <div
    class="w-full h-fit p-px bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500"
  >
    <div>
      <img
        alt="Goku"
        class="w-full"
        src="https:"
      />
      <p
        class="bg-slate-100 px-4 py-7 text-base text-center font-bold"
      >
        Goku
      </p>
    </div>
  </div>
</div>
`;
